volatility3/framework/plugins/windows/psxview.py

```diff
@@ -0,0 +1,279 @@
+                    if p.has_member("ObjectTable"):
+                        handles_plugin = handles.Handles(
+                            context=self.context, config_path=self.config_path
+                        )
+                        hndls = list(handles_plugin.handles(p.ObjectTable))
+                        for h in hndls:
+                            if (
+                                h.get_object_type(
+                                    handles_plugin.get_type_map(
+                                        self.context, layer_name, symbol_table
+                                    )
+                                )
+                                == "Process"
+                            ):
+                                ret.append(h.Body.cast("_EPROCESS"))
+
+                except exceptions.InvalidAddressException:
+                    vollog.log(
+                        constants.LOGLEVEL_VVV, "Cannot access eprocess object table"
+                    )
+
+        ret = self.filter_garbage_procs(ret)
+        return self.proc_list_to_dict(ret)
+
+    def check_session(self, pslist_procs):
+        procs = [p for p in pslist_procs if p.get_session_id() != None]
+
+        return self.proc_list_to_dict(procs)
+
+    def _generator(self):
+        kernel = self.context.modules[self.config["kernel"]]
+
+        layer_name = kernel.layer_name
+        symbol_table = kernel.symbol_table_name
+
+        kdbg_list_processes = list(
+            pslist.PsList.list_processes(
+                context=self.context, layer_name=layer_name, symbol_table=symbol_table
+            )
+        )
+
+        processes = {}
+
+        processes["pslist"] = self.check_pslist(kdbg_list_processes)
+        processes["psscan"] = self.check_psscan(layer_name, symbol_table)
+        processes["thrdscan"] = self.check_thrdscan()
+        processes["csrss"] = self.check_csrss_handles(
+            kdbg_list_processes, layer_name, symbol_table
+        )
+        processes["sessions"] = self.check_session(kdbg_list_processes)
+
+        seen_offsets = set()
+        for source in processes:
+            for offset in processes[source]:
+                if offset not in seen_offsets:
+                    seen_offsets.add(offset)
+                    proc = processes[source][offset]
+
+                    pid = proc.UniqueProcessId
+                    name = self.proc_name_to_string(proc)
+
+                    exit_time = proc.get_exit_time()
+                    if type(exit_time) != datetime.datetime:
+                        exit_time = ""
+                    else:
+                        exit_time = str(exit_time)
+
+                    in_sources = {
+                        src: str(offset in processes[src]) for src in processes
+                    }
+
+                    if self.config["identify-expected"]:
+                        f = "False"
+                        n = "Normal"
+
+                        if in_sources["pslist"] == f:
```

Comment:
This is checking whether in_sources contains the string "False"?  This is a pretty weird way of testing things, I think I'd feel happier if these could be split out into checks for each process gathering method.

Reply:
I think I see your point. I'll respond to it in another comment getting at the same issue.